client/webserver/site/src/html/forms.tmpl

```diff
@@ -79,6 +81,8 @@
 {{end}}
 
 {{define "unlockWalletForm"}}
+{{$isRegisterPage := eq .Title "Register | Decred DEX"}}
+{{$passwordIsCached := .UserInfo.PasswordIsCached}}
```

Comment:
We really want to avoid string matching here. How about a two-argument `templateFuncs` function?

#### Or even better...

The wallet unlock form is one of the last forms in **forms.tmpl** that doesn't have a javascript class. It is still bound with a function (which is how they all were at one point) called `bindOpenWallet`. Maybe you can create the class (see others in **forms.js**) and create a method to hide the PW input and submit button, and then just hide them in code. We can do that, because this form is never displayed by default on a page, so the user will never see the elements flash on screen before being hidden. A major benefit of doing it that way is that if the user caches their password using a different form on the same page, we'll have the machinery in place to hide these fields without a page reload.

Reply:
Yeah that's true.. it's better not to have to reload the screen. I'll just try and do the whole thing without using the template arguments. Then, if the the client ever gets a key from the UI that cannot decode a password, the cookie will be cleared.